src/niscope/templates/session.py/fancy_fetch_array_measurement.py.mako

```diff
@@ -17,12 +17,8 @@
             self._fetch_num_records = -1 if num_records is None else num_records
             self._fetch_meas_num_samples = -1 if meas_num_samples is None else meas_num_samples
 
-        if isinstance(self._interpreter, _library_interpreter.LibraryInterpreter):
-            if meas_wfm_size is None:
-                meas_wfm_size = self._actual_meas_wfm_size(array_meas_function)
-        else:
-            if meas_wfm_size is not None:
-                raise ValueError('The argument "meas_wfm_size" must be None when using gRPC.')
+        if meas_wfm_size is None:
+            meas_wfm_size = self._actual_meas_wfm_size(array_meas_function)
 
         meas_wfm, wfm_info = self._${f['python_name']}(array_meas_function, meas_wfm_size, timeout)
```

Comment:
This adds a network round trip when `meas_wfm_size` is `None`. If you leave `FetchArrayMeasurementRequest.meas_wfm_size` unset or set it to -1, the server will call `ActualMeasWfmSize` for you.

https://github.com/ni/grpc-device/blob/main/source/custom/niscope_service.custom.cpp#L231